dbcon/joblist/tuplehashjoin.cpp

```diff
@@ -1921,15 +1929,17 @@ void TupleHashJoinStep::segregateJoiners()
     // Debugging code, this makes all eligible joins disk-based.
     else {
     	cout << "making all joins disk-based" << endl;
+		joinIsTooBig = true;
     	for (i = 0; i < smallSideCount; i++) {
-    		joinIsTooBig = true;
+            joiner[i]->setConvertToDiskJoin();
     		djsJoiners.push_back(joiners[i]);
     		djsJoinerMap.push_back(i);
     	}
     	return;
     }
     #endif
 
+    boost::mutex::scoped_lock sl(djsLock);
```

Comment:
Why do we take lock here?

Reply:
the lock protects djs and djsJoiners.  Abort() needs a consistent view of those, and it can come asynchronously.